soroban-env-host/src/test/token.rs

```diff
@@ -288,6 +348,48 @@ fn test_smart_token_init_and_balance() {
     );
 }
 
+#[test]
+fn test_smart_tokens_dont_support_classic_ops() {
+    let test = TokenTest::setup();
+    let admin = TestSigner::Ed25519(&test.admin_key);
+    let token = test.default_smart_token(&admin);
+    let account_id = signer_to_account_id(&test.host, &test.user_key);
+    let user = TestSigner::account(&account_id, vec![&test.user_key]);
+    test.create_classic_account(
+        &account_id,
+        vec![(&test.user_key, 100)],
+        10_000_000,
+        0,
+        [1, 0, 0, 0],
+        None,
+        None,
+    );
+
+    token
+        .mint(
+            &admin,
+            token.nonce(admin.get_identifier(&test.host)).unwrap(),
+            user.get_identifier(&test.host),
+            BigInt::from_u64(&test.host, 100_000).unwrap(),
+        )
+        .unwrap();
+
+    assert!(token
+        .export(
+            &user,
+            token.nonce(user.get_identifier(&test.host)).unwrap(),
+            1
+        )
+        .is_err());
```

Comment:
💡 Could we change the `.is_err()` to do an assertion on the `Err(value)`? Tests that check for the presence of an error can rot over time because an error may start occurring that has nothing to do with what the test is happening, but the test won't fail and alert us to that.

Reply:
Actually, I guess `HostError` is not comparable with itself because it doesn't implement `ParitalEq`, because deep down it contains `RawVal` values that are not comparable. 

@graydon This is another example of where `RawVal` lacking equality limits us from writing more specific tests.

Reply:
Sure, I've already updated the tests, the PR was just blocked by this PR and is blocked by https://github.com/stellar/rs-soroban-env/pull/532 (maybe I can include the tests there though).

I had to extract status manually, so I agree that some functionality for comparison would help. OTOH, without a meaningful debug representation it wouldn't be too useful. While writing these tests I realized that I would much rather convert host value to a regular value and use `assert_eq` on it (like I do here with e.g. `to_vec`), rather than doing an equality check on the host values with `==` and without a meaningful error message in case of a failure.